csv_consolidator.py

```diff
@@ -0,0 +1,61 @@
+import sys
+import re
+
+def consolidateCSV(csvPath, outputFile):
+    """ Takes the csv file located at csvPath that was downloaded from 
+    https://api.bitcoincharts.com/v1/csv/ and consolidates the duplicate 
+    timestamps into one entry with the price and volume values being the 
+    average of all entries with that timestamp. It is required that the 
+    csv file located at csvPath has the form: 
+        [timestamp], [price], [volume]
+    """
+    print("beginning consolidation of " + csvPath + " to output file " + \
+          outputFile + "...\nthis may take a while...\n")
+
+    with open(csvPath) as input:
+        firstLine = input.readline().split(",")
+        currentTimeStamp = firstLine[0]
+        price_sum = float(firstLine[1])
+        volume_sum = float(firstLine[2])
+        with open(outputFile, "w") as output:
```

Comment:
It looks like there is some redundancy in variable declaration.  The local variables are set to initial values, and then sometimes reset to new values in an iterative loop.  It seems like initially setting variables to a generic value, like 0, and then doing all file reading in the same place would make more sense.

Reply:
Thanks for the feedback! The reason why I did it this way is because I needed to do some arithmetic in order to efficiently get the average values. In order to do this I needed to have a running sum that was initialized to the initial values and then continue to increment them until the timestamp was different, after which I would divide the values by how many iterations there were, thus giving me the average. I'm normally a huge fan of tiny functions, but I couldn't think of any way that I could shorten these without having a function with (in my opinion) too many parameters.  It took a bit of messing around before settling on this decision.